src/main/java/sparksoniq/spark/iterator/flowr/ForClauseSparkIterator.java

```diff
@@ -54,30 +54,25 @@
 
     private static final long serialVersionUID = 1L;
     private String _variableName; // for efficient use in local iteration
-    private RuntimeIterator _expression;
+    private RuntimeIterator _assignmentIterator;
     private Map<String, DynamicContext.VariableDependency> _dependencies;
     private DynamicContext _tupleContext; // re-use same DynamicContext object for efficiency
     private FlworTuple _nextLocalTupleResult;
     private FlworTuple _inputTuple; // tuple received from child, used for tuple creation
 
     public ForClauseSparkIterator(
             RuntimeTupleIterator child,
-            VariableReferenceIterator variableReference,
-            RuntimeIterator assignmentExpression,
+            String variableName,
```

Comment:
Building an iterator just to get the name of the variable was redundant for 'for' and 'let' clauses. We simply pass in the variable name now

Reply:
Outdated